src/org/ods/orchestration/util/Project.groovy

```diff
@@ -1980,4 +2018,26 @@ class Project {
         }
     }
 
+    void addDefaults(String component) {
+        def rmURL = git.getOriginUrl()
+        def gitURL = this.getGitURLFromPath(this.steps.env.WORKSPACE, 'origin')
+        def actualUrl = gitURL.resolve("${this.key.toLowerCase()}-${component}.git").toString()
+        def repo = [
+            id: component,
+            data: [
+                openshift: [:],
+                documents: [:],
+            ],
+            include: false,
+            type: MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_CODE,
+            url: actualUrl,
+            branch: 'master'
+        ]
+
+        if (rmURL == repo.url) {
```

Comment:
Please, relocate this check so that we don't create the repo structure, if unneeded. Also, can't we check whether the component name matches the RM component name instead of comparing URL's?